docker-compose.yml

```diff
@@ -0,0 +1,41 @@
+version: '2'
+
+services:
+  nginx:
+    image: bitnami/nginx:1.12.1-r1
+    volumes:
+      - ./dev/nginx:/bitnami/nginx/conf/vhosts
+    depends_on:
+      - backend
+      - frontend
+    environment:
+      - VIRTUAL_HOST=fortress.docker
+      - VIRTUAL_PORT=8080
+    ports:
+      - 8080:8080
+  mongo:
```

Comment:
we need the backend to wait for MongoDB, so we get rid of this initial (eventual) crash:

```
backend_1   |
backend_1   | /app/node_modules/mongodb/lib/mongo_client.js:421
backend_1   |           throw err
backend_1   |           ^
backend_1   | MongoError: failed to connect to server [mongo:27017] on first connect [MongoError: connect ECONNREFUSED 172.19.0.2:27017]
```

We could use the `healthcheck` feature in docker compose 2.1. For instance, we can check:

```
  mongo:
  ...
    healthcheck:
        test: "/opt/bitnami/mongodb/bin/mongo admin --eval 'db.getUsers()'"
        interval: 5s
        retries: 20
```

and force the `backend` service to depend on the `mongo` service:

```
  backend:
  ...
    depends_on:
        mongo:
            condition: service_healthy
```

Reply:
Cool! I didn't know about the health check clause. I'll add it later

Reply:
Done